sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/PocoEntityValueBinder.cs

```diff
@@ -36,13 +37,21 @@ internal class PocoEntityValueBinder<TElement> : IValueBinder
 
         public async Task<object> GetValueAsync()
         {
-            var conversionResult = await _entityToPocoConverter(_originalEntity, null, _valueBindingContext).ConfigureAwait(false);
-            // When bound to TableEntity the _entityToPocoConverter will no-op
-            // when it happens make a copy of entity so the change tracking still works
-            if (ReferenceEquals(conversionResult, _originalEntity))
+            // Create a copy of the original entity so that we can properly
+            // track changes to byte arrays. This also handles the case
+            // where the customer binds directly to TableEntity as we return
+            // a new instance with copied values.
+            var copiedEntity = new TableEntity(_originalEntity);
+            foreach (var key in copiedEntity.Keys)
             {
-                return new TableEntity(_originalEntity);
+                if (copiedEntity[key] is byte[] bytes)
+                {
+                    var copy = new byte[bytes.Length];
```

Comment:
would it be possible to rent and return from the array pool and then use the span based APIs?

Reply:
Hmm, I don't think we can, because we have to return the copied bytes to the user and they can do whatever they want with it (if I'm understanding correctly).